lib/ansible/modules/crypto/get_certificate.py

```diff
@@ -161,11 +177,35 @@ def main():
         if not isfile(ca_cert):
             module.fail_json(msg="ca_cert file does not exist")
 
-    try:
-        cert = get_server_certificate((host, port), ca_certs=ca_cert)
-        x509 = crypto.load_certificate(crypto.FILETYPE_PEM, cert)
-    except Exception as e:
-        module.fail_json(msg="Failed to get cert from port with error: {0}".format(e))
+    if proxy_host:
+        try:
+            connect = "CONNECT %s:%s HTTP/1.0\r\n\r\n" % (host, port)
+            sock = socket()
+            atexit.register(sock.close)
+            sock.connect((proxy_host, proxy_port))
+            sock.send(connect.encode())
+            sock.recv(8192)
+
+            ctx = create_default_context()
+            ctx.check_hostname = False
+            ctx.verify_mode = CERT_NONE
+
+            if ca_cert:
+                ctx.verify_mode = CERT_OPTIONAL
+                ctx.load_verify_locations(cafile=ca_cert)
+
+            cert = ctx.wrap_socket(sock, server_hostname=host).getpeercert(True)
+            cert = DER_cert_to_PEM_cert(cert)
+            x509 = crypto.load_certificate(crypto.FILETYPE_PEM, cert)
```

Comment:
Why not use `crypto.FILETYPE_ASN1` here and remove the above call to `DER_cert_to_PEM_cert`?

Reply:
because I wanted to do it the same way as the following code: 

https://github.com/ansible/ansible/blob/60b4891e46920c0383bea006cf871cbd8dd4c5d5/lib/ansible/modules/crypto/get_certificate.py#L206

I think it’s better to use `crypto.FILETYPE_ASN1`, so I ’ll fix it.

Reply:
When I checked the code again, I needed the result of cert converted to a string.

https://github.com/ansible/ansible/blob/c4c8f2868aa0221c18cb988543f25c0c064ace0a/lib/ansible/modules/crypto/get_certificate.py#L211

Therefore, I think that PEM conversion processing by `DER_cert_to_PEM_cert` is necessary.

https://github.com/ansible/ansible/blob/c4c8f2868aa0221c18cb988543f25c0c064ace0a/lib/ansible/modules/crypto/get_certificate.py#L199

Thank you

Reply:
Ok, in that case, keep it :)